infrastructure/api/sse-api.yml

```diff
@@ -317,10 +318,47 @@ Resources:
     Properties:
       Handler: backend/api/src/handlers/auth/update-client.updateClientInRegistryHandler
       Description: Updates a client using an HTTP endpoint
+      KmsKeyArn: !Ref EnvironmentEncryptionKey
       Environment:
         Variables:
           AUTH_REGISTRATION_BASE_URL: !Ref AuthRegistrationBaseUrl
           AUTH_API_KEY: "{{resolve:ssm:/self-service/api/auth-api-key}}"
+      Policies:
+        - Statement:
+            - Effect: Allow
+              Action:
+                - "kms:Decrypt"
+                - "kms:Encrypt"
+              Resource:
+                - !Ref EnvironmentEncryptionKey
+  
+ # KMS configuration for auth kms key
+  EnvironmentEncryptionKey:
+    Type: AWS::KMS::Key
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+    # checkov:skip=CKV_AWS_7: "Ensure rotation for customer created CMKs is enabled"
+    Properties:
+      Description: Key for auth key encrypt/decrypt
+      Enabled: TRUE
+      KeyPolicy:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Action: kms:*
+            Resource: '*'
```

Comment:
✅ This is the default key policy
https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-default.html